components/gitpod-db/src/typeorm/entity/db-cost-center.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the Gitpod Enterprise Source Code License,
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { Entity, Column, PrimaryColumn, Index } from "typeorm";
+import { CostCenter } from "@gitpod/gitpod-protocol";
+import { TypeORM } from "../typeorm";
+import { Transformer } from "../transformer";
+
+@Entity()
+// on DB but not Typeorm: @Index("ind_lastModified", ["_lastModified"])   // DBSync
+export class DBCostCenter implements CostCenter {
+    @PrimaryColumn("uuid")
+    id: string;
+
+    @Column(TypeORM.UUID_COLUMN_TYPE)
+    @Index("ind_userId")
+    userId: string;
+
+    @Column({
+        ...TypeORM.UUID_COLUMN_TYPE,
+        default: "",
+        transformer: Transformer.MAP_EMPTY_STR_TO_UNDEFINED,
+    })
+    teamId?: string;
```

Comment:
IMO, this mapping should be opaque. As in, the column should not state it maps to a Team, but stay as `costID`. For now, that can contain the current format of `team:<teamID>` or `user:<userID>` but it allows to keep the cost center idea a bit more general than just teams.

Reply:
I've no problem with that, but we tend to use another pattern in other places and I'd prefer to stay consistent. E.g. Project?.teamId/Project?.userId

Reply:
The point is that the value shouldn't necessarily represent just a teamID, it could represent some other structure, ie `userId` when the attribution is to a user.

Reply:
Got that. If the `teamId` is null, the `userId` is the attribution target. 
If we plan to introduce a different type of attribution target, we'd add a new column for that type. This is the way we've extended entities so far.

Reply:
@AlexTugarev We already use the [`AttributionId`](https://github.com/gitpod-io/gitpod/blob/c7cfb42d785897fec9a9c2e4d7514e6906f92802/components/gitpod-protocol/src/attribution.ts#L19) in various places and APIs now. I think it makes sense to re-use it here as well. :+1: